src/extensions/LensExtension.js

```diff
@@ -0,0 +1,129 @@
+import { LayerExtension } from '@deck.gl/core';
+import lens from './lens-module';
+
+/**
+ * This deck.gl extension allows for a lens that selectively shows one channel in its chosen color and then the others in white.
+ * @param {Object} opts
+ * @param {boolean=} opts.isLensOn Whether or not to use the lens.
+ * @param {number=} opts.lensSelection Numeric index of the channel to be focused on by the lens.
+ * @param {number=} opts.lensRadius Pixel radius of the lens (default: 100).
+ * @param {Array.<number>=} opts.lensBorderColor RGB color of the border of the lens (default [255, 255, 255]).
+ * @param {number=} opts.lensBorderRadius Percentage of the radius of the lens for a border (default 0.02).
+ * */
+
+export default class LensExtension extends LayerExtension {
+  getShaders() {
+    return {
+      ...super.getShaders(),
+      modules: [lens]
+    };
+  }
+
+  initializeState(context, extension) {
+    const layer = this.getCurrentLayer();
+    // eslint-disable-next-line no-param-reassign
+    const onMouseMove = () => {
```

Comment:
This will run in five layers, including three composite layers. I would suggest an early exit in the absence of `this.model`.

Reply:
Good call.  I'm going to use `isComposite`